### PR TITLE
Event optimize cleanup

### PR DESCRIPTION
**Pull request by devbhakt**
Removed obsolete code that is no longer being used within event_optimize
- Manually tracking the progress of the MCMC run has been removed since the emcee sampler has a built in progress bar
- Global tracking of the maximum posterior has been removed as it was not working properly with parallel processing and the maximum posterior values were switched to being calculated using the metadata within the emcee sampler in the original parallel processing implementation. 
- Removed the custom timing class as it is not being used

Moved the portion of the code that generates a TOA object and weights from the photon data into a dedicated function.
- The function loads in the photon data, creates a TOAs object using the photon data between the passed min and max MJD arguments, and the corresponding photon weights. 
- Removed the usepickle argument as the function checks the events file name to determine it is a pickle file or not.

### CODE REVIEW

**Comment by abhisrkckl** [comment]
Please add a CHANGELOG entry

**Comment by abhisrkckl** [comment]
Can you please check why the documentation build is failing?

**Comment by abhisrkckl** [comment]
I think the issue is having the line ---- above the heading in the docstring (like Parameter) rather than below it.

**Comment by devbhakt** [comment]
Should be fixed now! Thanks @abhisrkckl

**Comment by devbhakt** [comment]
All tests have passed and the PR is ready for review

File: src/pint/scripts/event_optimize.py [hunk 1]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
     return autocorr
 
 
+def load_events_weights(eventfile, model, weightcol, wgtexp, minMJD, maxMJD, minWeight):
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:311]: Please add a test for this function.

File: src/pint/scripts/event_optimize.py [hunk 2]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+        wgtexp (float): Raise computed weights to this power (or 0.0 to disable any rescaling of weights)
+        minMJD (float): Earliest MJD to use
+        maxMJD (float): Latest MJD to use
+        minWeight (float)): Minimum weight to include
+    Returns
+    -------
+        ts: TOA object containing all of the photon data
+        weights: numpy array containing the corresponding weights
+    """
+    if "ELONG" in model.params:
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:328]: Isn't it better to check `"AstrometryEcliptic" in model.components`?

File: src/pint/scripts/event_optimize.py (hunk 1 above)
Comment (abhisrkckl) [reply src/pint/scripts/event_optimize.py:311]: Also, can you please add type hints here?

File: src/pint/scripts/event_optimize.py [hunk 3]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+            minmjd=minMJD,
+            maxmjd=maxMJD,
+            ephem="DE421",
+            planets=False,
+        )
+        ts.filename = eventfile
+        try:
+            toa.save_pickle(ts)
+        except IOError:
+            pass
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:363]: Please use `contextlib.suppress`

File: src/pint/scripts/event_optimize.py [hunk 4]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+    target = tc if weightcol == "CALC" else None
+
+    ts = None
+    if eventfile.endswith("pickle") or eventfile.endswith("pickle.gz"):
+        try:
+            ts = toa.load_pickle(eventfile)
+            mjds = ts.get_mjds().value
+            ts = ts[(mjds >= minMJD) & (mjds <= maxMJD)]
+        except IOError:
+            pass
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:346]: Please use `contextlib.suppress`

File: src/pint/scripts/event_optimize.py [hunk 5]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+            for ii, x in enumerate(ts.table["flags"]):
+                x["weight"] = str(weights[ii])
+        weights = np.asarray([float(x["weight"]) for x in ts.table["flags"]])
+
+        ts = ts[weights >= minWeight]
+        weights = weights[weights >= minWeight]
+        log.info(
+            "There are %d events, with min / max weights %.3f / %.3f"
+            % (len(weights), weights.min(), weights.max())
+        )
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:387]: Please use fstrings

File: src/pint/scripts/event_optimize.py [hunk 6]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+    else:
+        weights = None
+        log.info("There are %d events, no weights are being used." % ts.ntoas)
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:390]: Please use fstrings

File: src/pint/scripts/event_optimize.py [hunk 7]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+        try:
+            toa.save_pickle(ts)
+        except IOError:
+            pass
+
+    if weightcol is not None:
+        if weightcol == "CALC":
+            weights = np.asarray([float(x["weight"]) for x in ts.table["flags"]])
+            log.info(
+                "Original weights have min / max weights %.3f / %.3f"
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:369]: Please use fstrings

File: src/pint/scripts/event_optimize.py [hunk 8]
```diff
@@ -323,6 +308,90 @@ def run_sampler_autocorr(sampler, pos, nsteps, burnin, csteps=100, crit1=10):
...
+                # make the highest weight = 1, but keep min weight the same
+                weights = wmn + ((weights - wmn) * (1.0 - wmn) / (wmx - wmn))
+            for ii, x in enumerate(ts.table["flags"]):
+                x["weight"] = str(weights[ii])
+        weights = np.asarray([float(x["weight"]) for x in ts.table["flags"]])
+
+        ts = ts[weights >= minWeight]
+        weights = weights[weights >= minWeight]
+        log.info(
+            "There are %d events, with min / max weights %.3f / %.3f"
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:385]: Please use fstrings

File: src/pint/scripts/event_optimize.py [hunk 9]
```diff
@@ -523,7 +582,7 @@ def main(argv=None):
         "--minMJD", help="Earliest MJD to use", type=float, default=54680.0
     )
     parser.add_argument(
-        "--maxMJD", help="Latest MJD to use", type=float, default=57250.0
+        "--maxMJD", help="Latest MJD to use", type=float, default=65000.0
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:585]: What is the origin of these defaults?

File: src/pint/scripts/event_optimize.py [hunk 10]
```diff
@@ -567,12 +626,6 @@ def main(argv=None):
         type=float,
         default=10.0,
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:627]: Why is this default out of [0,1]?

File: CHANGELOG-unreleased.md [hunk 11]
```diff
@@ -9,6 +9,25 @@ the released changes.
...
+- Type hinting for most of the `pint.utils` module
+- `funcParameter`s are no longer listed in the `pintk` interface.
+- Updated location of CCERA
+- Removed `include_bipm`, and `bipm_version` from the Observatory class. Now they are passed as arguments to `site.clock_correction()`
+- Renamed `include_gps` to `apply_gps2utc` in the Observatory class
+- Removed `apply_gps2utc` from `TOAs.clock_corr_info` since it can be different for different observatories. It is not a global setting.
+- The following observatories no longer have a default of `include_bipm=False`: magic, lst, virgo, lho, llo, geo600, kagra, hess, hawc
+- New algorithm for TCB <-> TDB conversion
+- Reordered plotting axes in `pintk`
+- Changed `scipy.integrate.simps` to `scipy.integrate.simpson` to work with scipy 1.14
```
Comment (abhisrkckl) [inline CHANGELOG-unreleased.md:25]: Please delete these lines. They have been moved from this file when we tagged the new version.

Comment (devbhakt) [reply CHANGELOG-unreleased.md]: Updated! Thanks for pointing that out

**Pull request merged**